### PR TITLE
New script: SnipeIT LXC

### PR DESCRIPTION
**Pull request by michelroegl-brunner**
> [!NOTE]
> We are meticulous when it comes to merging code into the main branch, so please understand that we may reject pull requests that do not meet the project's standards. It's never personal. Also, game-related scripts have a lower chance of being merged.

## Description

I have the Bare Metal install Script for SnipeIT up and running. 
You can test it on your Proxmox with this Script: https://raw.githubusercontent.com/michelroegl-brunner/ProxmoxVE/refs/heads/snipeit-bare-metal/ct/snipeit.sh (I´ve changed all links to my personal github for testing)

Thank you for your time for the Review in advance!

Fixes # (issue)

## Type of change
Please check the relevant option(s):

- [ ] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [X] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [X] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [X] Testing performed (I have tested my changes, ensuring everything works as expected)
- [ ] Documentation updated (I have updated any relevant documentation)

## Related Pull Requests / Discussions

If there are other pull requests or discussions related to this change, please link them here:
- Related PR #

### CODE REVIEW

File: install/snipeit-install.sh [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+update_os
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y curl
+$STD apt-get install -y git
+$STD apt-get install -y sudo
+$STD apt-get install -y mc
+$STD apt-get install -y nginx
+$STD apt-get install -y php8.2-{bcmath,common,ctype,curl,fileinfo,fpm,gd,iconv,intl,mbstring,mysql,soap,xml,xsl,zip,cli}
+$STD apt-get install -y mariadb-server
```
Comment (MickLesk) [inline install/snipeit-install.sh:22]:
```bash
$STD apt-get install -y \
  curl \
  git \
  sudo \
  mc \
  nginx \
  php8.2-{bcmath,common,ctype,curl,fileinfo,fpm,gd,iconv,intl,mbstring,mysql,soap,xml,xsl,zip,cli} \
  mariadb-server 
```

is the smarter way, is realy every of this php8.2 packages needed?

File: install/snipeit-install.sh [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+msg_ok "Installed Dependencies"
+
+msg_info "Installing Composer"
+curl -sS https://getcomposer.org/installer -o composer-setup.php
+php composer-setup.php --install-dir=/usr/local/bin --filename=composer
+msg_ok "Installed Composer"
```
Comment (MickLesk) [inline install/snipeit-install.sh:28]: why not the debian/ubuntu default composer with apt-get install -y composer?

File: install/snipeit-install.sh [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+msg_ok "Configre Database\n"
+read -r -p "Enter password for Database user: " password
+echo -e 'CREATE DATABASE snipeit;' | mysql
+echo -e "GRANT ALL ON snipeit.* TO snipeit@localhost identified by '$password';" | mysql
+echo -e 'FLUSH PRIVILEGES;' | mysql
+msg_ok "Configured Database"
```
Comment (MickLesk) [inline install/snipeit-install.sh:35]: The script breaks with our current conventions. Here is an example

```bash
msg_info "Setting up database"
DB_NAME=snipeit_db
DB_USER=snipeit
DB_PASS=$(openssl rand -base64 18 | tr -dc 'a-zA-Z0-9' | head -c13)
mysql -u root -e "CREATE DATABASE $DB_NAME;"
mysql -u root -e "CREATE USER '$DB_USER'@'localhost' IDENTIFIED WITH mysql_native_password AS PASSWORD('$DB_PASS');"
mysql -u root -e "GRANT ALL ON $DB_NAME.* TO '$DB_USER'@'localhost'; FLUSH PRIVILEGES;"
{
    echo "SnipeIT-Credentials"
    echo "SnipeIT Database User: $DB_USER"
    echo "SnipeIT Database Password: $DB_PASS"
    echo "SnipeIT Database Name: $DB_NAME"
} >> ~/snipeit.creds
msg_ok "Set up database"
```

File: install/snipeit-install.sh [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+msg_info "Clone SnipeIT from Github"
+cd /var/www/html
+git clone https://github.com/snipe/snipe-it snipe-it
+msg_ok "Finished cloning"
```
Comment (MickLesk) [inline install/snipeit-install.sh:40]: Why git clone? So do you have an full zip file from release state

```bash
msg_info "Installing Snipe-IT"
cd /opt
RELEASE=$(curl -s https://api.github.com/repos/snipe/snipe-it/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
wget -q "https://github.com/snipe/snipe-it/archive/refs/tags/v${RELEASE}.zip"
unzip -q v${RELEASE}.zip
mv snipe-it-${RELEASE} /opt/snipe-it
.
.
.
.
```

File: install/snipeit-install.sh [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+msg_info "Configure SnipeIT"
+cd snipe-it
+cp .env.example .env
+IPADDRESS=$(hostname -I | awk '{print $1}')
+sed -i "s|^APP_URL=.*|APP_URL=http://$IPADDRESS|" .env
+
+sed -i "s|^DB_DATABASE=.*|DB_DATABASE=snipeit|" .env
+sed -i "s|^DB_USERNAME=.*|DB_USERNAME=snipeit|" .env
+sed -i "s|^DB_PASSWORD=.*|DB_PASSWORD=$password|" .env
```
Comment (MickLesk) [inline install/snipeit-install.sh:50]:
```bash
sed -i -e "s|^APP_URL=.*|APP_URL=http://$IPADDRESS|" \
       -e "s|^DB_DATABASE=.*|DB_DATABASE=snipeit|" \
       -e "s|^DB_USERNAME=.*|DB_USERNAME=snipeit|" \
       -e "s|^DB_PASSWORD=.*|DB_PASSWORD=$password|" .env
```

File: ct/snipeit.sh [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+clear
+cat <<"EOF"
+   _____       _            _____ _______ 
+  / ____|     (_)          |_   _|__   __|
+ | (___  _ __  _ _ __   ___  | |    | |   
+  \___ \| '_ \| | '_ \ / _ \ | |    | |   
+  ____) | | | | | |_) |  __/_| |_   | |   
+ |_____/|_| |_|_| .__/ \___|_____|  |_|   
+                | |                       
+                |_|
```
Comment (MickLesk) [inline ct/snipeit.sh:17]: https://www.patorjk.com/software/taag/#p=display&f=Slant&t=Snipe-IT

**Review by MickLesk — CHANGES_REQUESTED** [review]
I have made a few comments on this. Many script sections do not fit our conventions. If you have the time and desire, you are welcome to take a look at current examples, such as adventurelog / komga / tandoor / listmonk and co. 

In case the question arises, unfortunately we don't have a template yet

File: install/snipeit-install.sh (hunk 1 above)
Comment (michelroegl-brunner) [reply install/snipeit-install.sh:22]: Yep, this is the better way. And yes, all this extentions are needed per requirements. (https://snipe-it.readme.io/docs/requirements)

File: install/snipeit-install.sh (hunk 2 above)
Comment (michelroegl-brunner) [reply install/snipeit-install.sh:28]: This was done following the documentation(https://snipe-it.readme.io/docs/install-dependencies), but it is a goo hint, i will try the defaukt composer.

File: install/snipeit-install.sh [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+msg_info "Clone SnipeIT from Github"
+cd /var/www/html
+git clone https://github.com/snipe/snipe-it snipe-it
+msg_ok "Finished cloning"
+
+msg_info "Configure SnipeIT"
+cd snipe-it
+cp .env.example .env
+IPADDRESS=$(hostname -I | awk '{print $1}')
```
[the remark this answers is not in the document]
Comment (michelroegl-brunner) [reply install/snipeit-install.sh:45]: Per my previous testing (bevor doing this script adn running it in production) this is necessery, yes.

File: install/snipeit-install.sh [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+sed -i "s|^APP_URL=.*|APP_URL=http://$IPADDRESS|" .env
+
+sed -i "s|^DB_DATABASE=.*|DB_DATABASE=snipeit|" .env
+sed -i "s|^DB_USERNAME=.*|DB_USERNAME=snipeit|" .env
+sed -i "s|^DB_PASSWORD=.*|DB_PASSWORD=$password|" .env
+chown -R www-data: /var/www/html/snipe-it
+chmod -R 755 /var/www/html/snipe-it
```
[the remark this answers is not in the document]
Comment (michelroegl-brunner) [reply install/snipeit-install.sh:52]: I have to test if it will work with nginx when the web related files are not in /var/www

**Comment by michelroegl-brunner** [comment]
Im working in your requested changes and i will have a good look on your examples listed above! 
Thank you for your time!

**Review by MickLesk — APPROVED** [review]

**Comment by havardthom** [comment]
Why has this new script been merged with 1 approval?

**Comment by shreksparrow** [comment]
i have installed snipe-it as i would any other LXC. then i set up a reverse proxy using nginx proxy manager but i get stuck after that. when i load the container on my local network via http://IP:80 i get redirected to /setup etc. yet when i reverse proxy the snipe-it LXC i get greeted by the Welcome to nginx! page. ive tried changin the app url to my https://domain, set forceSSL to true, tried anything really. why does this not work

**Comment by michelroegl-brunner** [comment]
> i have installed snipe-it as i would any other LXC. then i set up a reverse proxy using nginx proxy manager but i get stuck after that. when i load the container on my local network via http://IP:80 i get redirected to /setup etc. yet when i reverse proxy the snipe-it LXC i get greeted by the Welcome to nginx! page. ive tried changin the app url to my https://domain, set forceSSL to true, tried anything really. why does this not work

Can you please open an Issue for that? In the meantime i will have a look at it.

**Comment by michelroegl-brunner** [comment]
@shreksparrow 
Disregard the creation of the issue.

I´ve go i running and postet a guide: [SnipeIT with Reverse Proxy](https://github.com/community-scripts/ProxmoxVE/discussions/671)

**Pull request merged**